### PR TITLE
Consul integration

### PR DESCRIPTION
**Pull request by nvtkaszpir**
Document with example how to use consul in gluster ecosystem to make configuration easier and more resilient to failures. Linux and Windows.

### CODE REVIEW

File: Administrator Guide/Consul.md [hunk 1]
```diff
@@ -0,0 +1,607 @@
...
+
+[Consul](https://www.consul.io/) is used for service discovery and configuration.
+
+It consists of consul server and agents connecting to it.
+Apps can get configuration data from consul via HTTP API or DNS queries.
+
+# Warning
+
+In this document you will get a proof-of-concept basic setup - gluster servers and gluster clients configured,
+which should be a point to expand. You should read [Further steps](# Further steps for improvements) section to harden it.
```
Comment (prashanthpai) [inline Administrator Guide/Consul.md]: This link isn't rendered in GitHub properly. But that's ok.

File: Administrator Guide/Consul.md [hunk 2]
```diff
@@ -0,0 +1,607 @@
...
+```
+
+This is a bit inconvenient, for example we have an image with hardcoded hostnames, and old servers are gone due to maintenance.
+We would have to recreate image, or reconfigure existing nodes if they unmount gluster storage.
+
+To mitigate that issue we can now use consul for fetching the server pool:
+
+```bash
+
+    mount -t glusterfs gluster.service.consul:/g0 /mnt/gluster/g0
```
Comment (prashanthpai) [inline Administrator Guide/Consul.md]: Ah, I see. I'm new to consul and this is where I realise what this document is all about. Can you briefly add what consul integration solves for the users at the beginning of this doc ?

Thanks for the detailed PR 👍

Comment (nvtkaszpir) [inline Administrator Guide/Consul.md]: Good idea ,fixed in the following commits.

File: Administrator Guide/Consul.md (hunk 1 above)
Comment (nvtkaszpir) [inline Administrator Guide/Consul.md]: Good catch. Fixed

**Pull request merged**